File: Leetcode/PY/Leetcode594-最长和谐子序列.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        nums.sort()
        l,r,res = 0,0,0
        while r < len(nums):
            while nums[r] - nums[l] > 1: l += 1
            if nums[r] - nums[l] == 1:
                res = max(res,r - l + 1)
            r += 1
        return res

File: Leetcode/Rust/Leetcode594-最长和谐子序列.rs
use std::collections::HashMap;
impl Solution {
    pub fn find_lhs(nums: Vec<i32>) -> i32 {
        let mut mp = HashMap::new();
        for x in nums.iter(){
            *mp.entry(x).or_insert(0) += 1;
        }
        let mut res = 0;
        for (&k,&v) in mp.iter(){
            match mp.get(&(k - 1)){
                Some(x) => res = res.max(x + v),
                None => {},
            }
        } 
        res
    }
}